rpc-server/src/modules/blocks/methods.rs

```diff
@@ -19,48 +20,94 @@ pub async fn block(
     Params(mut params): Params<near_jsonrpc_primitives::types::blocks::RpcBlockRequest>,
 ) -> Result<near_jsonrpc_primitives::types::blocks::RpcBlockResponse, RPCError> {
     tracing::debug!("`block` called with parameters: {:?}", params);
-
-    // Increase the OPTIMISTIC_REQUESTS_TOTAL metric if the request has optimistic finality.
-    if let near_primitives::types::BlockReference::Finality(finality) = &params.block_reference {
-        // Finality::None stands for optimistic finality.
-        if finality == &near_primitives::types::Finality::None {
-            crate::metrics::OPTIMISTIC_REQUESTS_TOTAL.inc();
-        }
-    }
     crate::metrics::BLOCK_REQUESTS_TOTAL.inc();
-    let result = fetch_block(&data, params.block_reference.clone()).await;
-
-    #[cfg(feature = "shadow_data_consistency")]
-    {
-        let near_rpc_client = data.near_rpc_client.clone();
-        let error_meta = format!("BLOCK: {:?}", params);
-        let read_rpc_response_json = match &result {
-            Ok(res) => {
-                if let near_primitives::types::BlockReference::Finality(_) = params.block_reference
+    match &params.block_reference {
+        near_primitives::types::BlockReference::SyncCheckpoint(_) => {
+            crate::metrics::SYNC_CHECKPOINT_REQUESTS_TOTAL.inc();
+            let block_view = proxy_rpc_call(&data.near_rpc_client, params).await?;
+            Ok(near_jsonrpc_primitives::types::blocks::RpcBlockResponse { block_view })
+        }
+        near_primitives::types::BlockReference::Finality(finality) => {
+            if finality != &near_primitives::types::Finality::Final {
```

Comment:
I think it'd be more honest to compare `finality` with `None` here

Reply:
We supports only Finality::Final. In addition to Finality::None we have an additional case of Finality::DoomSlug, which should also be proxied